picoquic/logwriter.c

```diff
@@ -746,7 +748,8 @@ FILE* create_binlog(char const* binlog_file, uint64_t creation_time);
 
 void binlog_new_connection(picoquic_cnx_t * cnx)
 {
-    if (cnx == NULL || cnx->quic->binlog_dir == NULL) {
+    char const* bin_dir = (cnx->quic->binlog_dir == NULL) ? cnx->quic->qlog_dir : cnx->quic->binlog_dir;
+    if (cnx == NULL || bin_dir == NULL) {
```

Comment:
If cnx was NULL then you'd have crashed the line before already :)

Reply:
Removed the extra test. It would mess up static analysis.